packages/codegen-ui-react/lib/react-component-render-helper.ts

```diff
@@ -70,6 +72,10 @@ export function isBoundProperty(prop: ComponentPropertyValueTypes): prop is Boun
   return 'bindingProperties' in prop;
 }
 
+export function isAuthProperty(prop: ComponentPropertyValueTypes): prop is StudioComponentAuthProperty {
+  return 'userAttribute' in prop;
+}
+
```

Comment:
Do you want to push this up to the renderHelper, since you basically reimplement this check in renderer-helper inside the `hasAuth` check.

Reply:
Good suggestion, but more changes/refactor but I too feel its worth doing it now so its helpful in future :)

Reply:
Done

Reply:
I think this needs to move to renderer-helper class, codegen-ui doesn't depend on codegen-ui-react, so I'm not sure how this'll build as-is.